cmd/courier/watch_test.go

```diff
@@ -0,0 +1,34 @@
+package courier
+
+import (
+	"context"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/ory/kratos/internal"
+	"github.com/stretchr/testify/require"
+)
+
+func TestStartCourier(t *testing.T) {
+	t.Run("case=without metrics", func(t *testing.T) {
+		ctx, cancel := context.WithCancel(context.Background())
+		_, r := internal.NewFastRegistryWithMocks(t)
+		go StartCourier(ctx, r)
+		require.Equal(t, r.Config(ctx).CourierExposeMetricsPort(), 0)
+		cancel()
+
+	})
+	t.Run("case=with metrics", func(t *testing.T) {
+		ctx, cancel := context.WithCancel(context.Background())
+		_, r := internal.NewFastRegistryWithMocks(t)
+		r.Config(ctx).Set("expose-metrics-port", 8080)
```

Comment:
@mattbonnell @mbonnell-wish this fails in the CI because port 8080 is allocated (see https://app.circleci.com/pipelines/github/ory/kratos/2818/workflows/0760a6cb-6853-4c3d-bc92-718cfd897e28/jobs/18333). You can use the freeport package to find a free port!

Reply:
fixed! (on my branch)